PayPalNative/src/test/java/com/braintreepayments/api/PayPalNativeClientTest.java

```diff
@@ -0,0 +1,759 @@
+        payPalNativeTokenizeCallback = mock(PayPalNativeTokenizeCallback.class);
+        payPalClient = mock(PayPalClient.class);
+        PowerMockito.mockStatic(PayPalCheckout.class);
+        PowerMockito.mockStatic(Uri.class);
+    }
+
+    @Test
+    public void tokenizePayPalAccount_whenPayPalNotEnabled_throwsError() {
+        TokenizationClient tokenizationClient = new MockTokenizationClientBuilder().build();
+        PayPalInternalClient payPalInternalClient = new MockPayPalInternalClientBuilder().build();
+
+        BraintreeClient braintreeClient = new MockBraintreeClientBuilder()
+                .configuration(payPalDisabledConfig)
+                .build();
+
+        PayPalNativeClient sut = new PayPalNativeClient(braintreeClient, tokenizationClient, payPalInternalClient, payPalClient);
+        sut.tokenizePayPalAccount(context, new PayPalNativeCheckoutRequest("1.0"), payPalNativeTokenizeCallback);
+
+        ArgumentCaptor<Exception> errorCaptor = ArgumentCaptor.forClass(Exception.class);
+        verify(payPalNativeTokenizeCallback).onResult(isNull(), errorCaptor.capture());
+        assertTrue(errorCaptor.getValue() instanceof BraintreeException);
+        assertEquals("PayPal is not enabled. " +
+                "See https://developers.braintreepayments.com/guides/paypal/overview/android/ " +
+                "for more information.", errorCaptor.getValue().getMessage());
+    }
+
+    @Test
+    public void tokenizePayPalAccount_whenDeviceCantPerformBrowserSwitch_throwsError() {
+        final String errorMessage = "AndroidManifest.xml is incorrectly configured or another app " +
+                "defines the same browser switch url as this app. See " +
+                "https://developers.braintreepayments.com/guides/client-sdk/android/#browser-switch " +
+                "for the correct configuration";
+        TokenizationClient tokenizationClient = new MockTokenizationClientBuilder().build();
+        PayPalInternalClient payPalInternalClient = new MockPayPalInternalClientBuilder().build();
+
+        BraintreeClient braintreeClient = new MockBraintreeClientBuilder()
+                .configuration(payPalEnabledConfig)
+                .canPerformBrowserSwitch(false)
+                .build();
+
+        doAnswer(new Answer<Void>() {
+            @Override
+            public Void answer(InvocationOnMock invocation) throws Throwable {
+                PayPalFlowStartedCallback callback = invocation.getArgument(2, PayPalFlowStartedCallback.class);
+                callback.onResult(new BraintreeException(errorMessage));
+
+                ArgumentCaptor<Exception> errorCaptor = ArgumentCaptor.forClass(Exception.class);
+                verify(payPalNativeTokenizeCallback).onResult(isNull(), errorCaptor.capture());
+                assertTrue(errorCaptor.getValue() instanceof BraintreeException);
+                assertEquals(errorMessage, errorCaptor.getValue().getMessage());
```

Comment:
For all of these tests lets move the test assertions outside of the mocking. I think these can all go after `sut.tokenizePayPalAccount` (this comment applies to all the tests in this file)

Reply:
Im going to apply it to other tests, but this one im going to delete it because it doesnt apply now